### ISSUE TITLE
Modifier keys for neo layout variants are not working

### ISSUE THREAD

**Issue by heldderarbeit** [wez/wezterm#3913]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

GNOME 44.2

### WezTerm version

wezterm 20230629-080200-f376468f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

I'm using a variant of the [Neo keyboard layout](https://www.neo-layout.org/), which works flawlessly in the regular [wezterm](https://archlinux.org/packages/extra/x86_64/wezterm/) package from the Arch Extra Repo. But in the [wezterm-git](https://aur.archlinux.org/packages/wezterm-git) package and the flatpak nightly, some of the modifier keys for Layer 3/4 are not working correctly anymore.

### To Reproduce

For example: AltGr key is a Mod4 key.

Mod4 + these keys:

![image](https://github.com/wez/wezterm/assets/4840355/f54c7bf8-2499-4a67-a189-1e7d63b8bb73)

should move the position of the cursor: 
![image](https://github.com/wez/wezterm/assets/4840355/96c7f42c-27a5-45a4-bb85-f6f6a69114fe)

In wezterm this works, but it doesnt in wezterm-git. Here it just produces regular Layer1 lower case letters.

Another example: right Mod3 key 

![image](https://github.com/wez/wezterm/assets/4840355/c5e34519-ffbf-45c8-ae90-760e79cb5527)

I want to print these symbols:

![image](https://github.com/wez/wezterm/assets/4840355/225b9cc3-2274-4ec3-bc07-c9d490dfdce2)

for each of the symbols, a " \ " is incorrectly prepended. So if I want to write "/", I get " \ / "

### Configuration

no config

**Comment by heldderarbeit** [comment]
With the help from [this guide](https://github.com/wez/wezterm/issues/3810#issuecomment-1571255662), I'm able to produce some logs.

Situation: I'm pressing Mod 3 plus this symbol to produce: / (forward slash)

![Untitled](https://github.com/wez/wezterm/assets/4840355/0fc4dcf3-26a6-4aac-8839-a360915dab2f)

`localectl status`
```
System Locale: LANG=en_IE.UTF-8
    VC Keymap: koy
   X11 Layout: de
    X11 Model: pc105
  X11 Variant: koy
```

I'm switching to x11 to use xev. But the problem occurs in both Wayland and X.

`xev -event keyboard` in gnome-terminal, then pressing these 2 keys together.

```
Outer window is 0x5200001, inner window is 0x5200002

KeymapNotify event, serial 24, synthetic NO, window 0x0,
    keys:  130 0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
           0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   

KeyPress event, serial 25, synthetic NO, window 0x5200001,
    root 0x75b, subw 0x0, time 89900909, (588,530), root:(688,699),
    state 0x0, keycode 51 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XLookupString gives 0 bytes: 
    XmbLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeyPress event, serial 28, synthetic NO, window 0x5200001,
    root 0x75b, subw 0x0, time 89902453, (588,530), root:(688,699),
    state 0x80, keycode 39 (keysym 0x2f, slash), same_screen YES,
    XLookupString gives 1 bytes: (2f) "/"
    XmbLookupString gives 1 bytes: (2f) "/"
    XFilterEvent returns: False

KeyRelease event, serial 28, synthetic NO, window 0x5200001,
    root 0x75b, subw 0x0, time 89902517, (588,530), root:(688,699),
    state 0x80, keycode 39 (keysym 0x2f, slash), same_screen YES,
    XLookupString gives 1 bytes: (2f) "/"
    XFilterEvent returns: False

KeyRelease event, serial 28, synthetic NO, window 0x5200001,
    root 0x75b, subw 0x0, time 89903677, (588,530), root:(688,699),
    state 0x80, keycode 51 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XLookupString gives 0 bytes: 
    XFilterEvent returns: False

KeymapNotify event, serial 28, synthetic NO, window 0x0,
    keys:  91  0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
           0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   0   
```

Now we are comparing `WEZTERM_LOG=window::os::x11::keyboard=trace,info wezterm -n --config debug_key_events=true` between different wezterm versions. 

First, the version from the arch extra repo (`20230408-112425-69ae8472`). Here everything is working as expected:
```
18:14:10.927  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(Backslash), modifiers: NONE, phys_code: Some(Backslash), raw_code: 51, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
18:14:10.929  TRACE  window::os::x11::keyboard         > process_key_event: RawKeyEvent FeedResult::Nothing: "", 65027. kc -> None
18:14:10.929  TRACE  window::os::x11::keyboard         > keysym_to_keycode for 65027 and 65027 -> None
18:14:13.190  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(S), modifiers: NONE, phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
18:14:13.190  TRACE  window::os::x11::keyboard         > process_key_event: RawKeyEvent FeedResult::Nothing: "/", 47. kc -> Some(Char('/'))
18:14:13.191  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: Char('/'), modifiers: NONE, repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(S), modifiers: NONE, phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: true, handled: Handled(false) }) } 
18:14:13.191  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=Char('/') mods=Modifiers(0x0)
18:14:13.191  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "/", Char('/') Modifiers(0x0)
18:14:13.238  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('/'), modifiers: NONE, repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(S), modifiers: NONE, phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: false, handled: Handled(false) }) } 
18:14:13.238  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char('/') mods=Modifiers(0x0)
18:14:14.664  TRACE  window::os::x11::keyboard             > keysym_to_keycode for 65027 and 65027 -> None
```

Now, a log for the latest AppImage Nightly (`20230709-073836-1bfaf852`). Here I get the following output: "backward slash, then forward slash" instead of a regular forward slash.

```
18:19:19.224  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(Backslash), modifiers: NONE, leds: (empty), phys_code: Some(Backslash), raw_code: 51, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
18:19:19.225  TRACE  window::os::x11::keyboard         > process_key_event: RawKeyEvent FeedResult::Nothing: "", 65027. kc -> None def_sym=92
18:19:19.225  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: Char('\\'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(Backslash), modifiers: NONE, leds: (empty), phys_code: Some(Backslash), raw_code: 51, repeat_count: 1, key_is_down: true, handled: Handled(false) }) } 
18:19:19.225  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=Char('\\') mods=NONE
18:19:19.229  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "\\", Char('\\') NONE
18:19:21.023  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Physical(S), modifiers: NONE, leds: (empty), phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
18:19:21.023  TRACE  window::os::x11::keyboard             > process_key_event: RawKeyEvent FeedResult::Nothing: "/", 47. kc -> Some(Char('/')) def_sym=115
18:19:21.024  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('/'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(S), modifiers: NONE, leds: (empty), phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: true, handled: Handled(false) }) } 
18:19:21.024  INFO   wezterm_gui::termwindow::keyevent     > send to pane DOWN key=Char('/') mods=NONE
18:19:21.024  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "/", Char('/') NONE
18:19:21.086  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('/'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(S), modifiers: NONE, leds: (empty), phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: false, handled: Handled(false) }) } 
18:19:21.086  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char('/') mods=NONE
18:19:22.216  TRACE  window::os::x11::keyboard             > keysym_to_keycode for 65027 and 65027 -> None
```

wezterm-git (20230709-073836-1bfaf852) from the AUR has the same problem as the AppImage Version.
``` 
18:49:07.659  INFO   wezterm_gui::termwindow::keyevent > key_event RawKeyEvent { key: Physical(Backslash), modifiers: NONE, leds: (empty), phys_code: Some(Backslash), raw_code: 51, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
18:49:07.659  TRACE  window::os::x11::keyboard         > process_key_event: RawKeyEvent FeedResult::Nothing: "", 65027. kc -> None def_sym=92
18:49:07.659  INFO   wezterm_gui::termwindow::keyevent > key_event KeyEvent { key: Char('\\'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(Backslash), modifiers: NONE, leds: (empty), phys_code: Some(Backslash), raw_code: 51, repeat_count: 1, key_is_down: true, handled: Handled(false) }) } 
18:49:07.660  INFO   wezterm_gui::termwindow::keyevent > send to pane DOWN key=Char('\\') mods=NONE
18:49:07.660  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "\\", Char('\\') NONE
18:49:09.527  INFO   wezterm_gui::termwindow::keyevent     > key_event RawKeyEvent { key: Physical(S), modifiers: NONE, leds: (empty), phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: true, handled: Handled(false) } 
18:49:09.528  TRACE  window::os::x11::keyboard             > process_key_event: RawKeyEvent FeedResult::Nothing: "/", 47. kc -> Some(Char('/')) def_sym=115
18:49:09.528  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('/'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: true, raw: Some(RawKeyEvent { key: Physical(S), modifiers: NONE, leds: (empty), phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: true, handled: Handled(false) }) } 
18:49:09.528  INFO   wezterm_gui::termwindow::keyevent     > send to pane DOWN key=Char('/') mods=NONE
18:49:09.528  INFO   wezterm_term::terminalstate::keyboard > key_down: sending "/", Char('/') NONE
18:49:09.567  INFO   wezterm_gui::termwindow::keyevent     > key_event KeyEvent { key: Char('/'), modifiers: NONE, leds: (empty), repeat_count: 1, key_is_down: false, raw: Some(RawKeyEvent { key: Physical(S), modifiers: NONE, leds: (empty), phys_code: Some(S), raw_code: 39, repeat_count: 1, key_is_down: false, handled: Handled(false) }) } 
18:49:09.568  INFO   wezterm_gui::termwindow::keyevent     > send to pane UP key=Char('/') mods=NONE
18:49:10.464  TRACE  window::os::x11::keyboard             > keysym_to_keycode for 65027 and 65027 -> None
```

**Comment by wez** [comment]
Likely a duplicate of:
* https://github.com/wez/wezterm/issues/3610

**Comment by wez** [comment]
This should be resolved now in `main`.

It typically takes about an hour before commits are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by heldderarbeit** [comment]
seems to have fixed everything in `20230709-122834-f09992f7`, Thanks!
